File: src/main/java/pl/zetosoftware/car/dto/CarFilterDto.java
package pl.zetosoftware.car.dto;

import lombok.Builder;
import pl.zetosoftware.car.enums.BodyTypeEnum;
import pl.zetosoftware.car.enums.TypeOfFuelEnum;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
public record CarFilterDto (String brand, String model, BigDecimal engineCapacity,
                            BodyTypeEnum bodyType, TypeOfFuelEnum typeOfFuel, Integer productionYear,
                            LocalDate freeFrom, LocalDate freeTo){

}

File: src/main/resources/carrentalapp/src/app/cars/car-filter-dto.ts
import { BodyType } from "./enums/body-type";
import { TypeOfFuel } from "./enums/type-of-fuel";

export class CarFilterDto {
    brand!: string;
    model!: string;
    engineCapacity!: number;
    bodyType!: BodyType;
    typeOfFuel!: TypeOfFuel;
    productionYear!: number;
    freeFrom!: Date;
    freeTo!: Date;
}
